### PR TITLE
[DROOLS-936] clean-up dependencies to get rid of duplicated classes

### PR DESCRIPTION
**Pull request by psiroky**

### CODE REVIEW

File: guvnor-m2repo-editor/guvnor-m2repo-editor-backend/pom.xml [hunk 1]
```diff
@@ -101,10 +107,10 @@
       <artifactId>kie-ci</artifactId>
     </dependency>
 
-    <dependency>
-      <groupId>javax.enterprise</groupId>
-      <artifactId>cdi-api</artifactId>
-    </dependency>
+    <!--<dependency>-->
```
Comment (manstis) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-backend/pom.xml]: @psiroky Want to keep this commented out for any reason?

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/pom.xml [hunk 2]
```diff
@@ -92,10 +92,10 @@
       <scope>provided</scope>
     </dependency>
 
-    <dependency>
-      <groupId>javax.enterprise</groupId>
-      <artifactId>cdi-api</artifactId>
-    </dependency>
+    <!--<dependency>-->
```
Comment (manstis) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/pom.xml]: Do you really want to keep the commented block?

File: guvnor-structure/guvnor-structure-backend/pom.xml [hunk 3]
```diff
@@ -92,10 +92,10 @@
       <artifactId>uberfire-commons-editor-api</artifactId>
     </dependency>
 
-    <dependency>
-      <groupId>javax.enterprise</groupId>
-      <artifactId>cdi-api</artifactId>
-    </dependency>
+    <!--<dependency>-->
```
Comment (manstis) [inline guvnor-structure/guvnor-structure-backend/pom.xml]: Commented block again...

Comment (psiroky) [inline guvnor-structure/guvnor-structure-backend/pom.xml]: Ouch my bad, I'll remove those.

**Pull request merged**